src/wallet/rpcwallet.cpp

```diff
@@ -4624,6 +4626,252 @@ UniValue generateBlocks(const JSONRPCRequest& request, std::shared_ptr<CReserveS
+
+    if (fShowSecrets )
+        EnsureWalletIsUnlocked(pwallet);
+
+    UniValue results(UniValue::VARR);
+
+    std::set<CStealthAddress>::iterator it;
+
+    for (it = pwallet->stealthAddresses.begin(); it != pwallet->stealthAddresses.end(); ++it)
+    {
+        if (it->scan_secret.size() < 1)
+            continue; // stealth address is not owned
+
+        UniValue entry(UniValue::VOBJ);
+
+        if (fShowSecrets)
+        {
+            entry.pushKV("Label        ", it->label);
+            entry.pushKV("Address      ", it->Encoded());
+            entry.pushKV("Scan Secret  ", HexStr(it->scan_secret.begin(), it->scan_secret.end()));
+            entry.pushKV("Spend Secret ", HexStr(it->spend_secret.begin(), it->spend_secret.end()));
+        }
+        else
+        {
+            entry.pushKV("Label        ", it->label);
+            entry.pushKV("Address      ", it->Encoded());
+        };
+
+        results.push_back(entry);
+    };
+
+    return results;
+}
+
+UniValue exportstealthaddress(const JSONRPCRequest& request)
+{
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = wallet.get();
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+   if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+            "exportstealthaddress <label/address>\n"
+            "\nExports the given stealth address.\n"
+            "\nArguments:\n"
+            "1. \"label\"   (string, optional, default=\"\") Stealth address label to export\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"label\": \"str\",          (string) Stealth address label.\n"
+            "  \"Scan Secret\": \"str\",    (string) Scan secret.\n"
+            "  \"Spend Secret\": \"str\",   (string) Spend secret.\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("exportstealthaddress", "mystealthaddress")
+            + HelpExampleRpc("exportstealthaddress", "mystealthaddress")
+        );
+
+    std::string stealth_address_label = request.params[0].get_str();
+
+    LOCK2(cs_main, pwallet->cs_wallet);
+
+    EnsureWalletIsUnlocked(pwallet);
+
+    UniValue result(UniValue::VOBJ);
+
+    std::set<CStealthAddress>::iterator it;
+    for (it = pwallet->stealthAddresses.begin(); it != pwallet->stealthAddresses.end(); ++it)
+    {
+        if (it->scan_secret.size() < 1)
+            continue; // stealth address is not owned
+
+        if (stealth_address_label == it->label || stealth_address_label == it->Encoded())
+        {
+            UniValue entry(UniValue::VOBJ);
+            entry.pushKV("Label        ", it->label);
+            entry.pushKV("Scan Secret  ", HexStr(it->scan_secret.begin(), it->scan_secret.end()));
+            entry.pushKV("Spend Secret ", HexStr(it->spend_secret.begin(), it->spend_secret.end()));
```

Comment:
those keys look kinda akward, maybe we should name them: `label`, `scan_secret`, `spend_secret`?